radsat-sk/operation/message/protobuf/proto/RFileTransfer.proto

```diff
@@ -1,37 +1,123 @@
+		float vurx_curr		= 6; 	///< Tx Telemetry receiver current.
+		float vupa_curr		= 7; 	///< Tx Telemetry power amplifier current.
+		float pa_temp		= 8; 	///< Rx Telemetry power amplifier temperature
+		float board_temp	= 9; 	///< Rx Telemetry board temperature.
+		uint32 uptime		= 10;	///< Rx Uptime in Seconds.
+		uint32 frames		= 11;	///< Rx Frames currently in Receive Buffer.
+	}
+
+	// Telemetry for the Transmitter
+	message transmitterTelemetry {
+		float tx_reflpwr	= 1; 	///< Tx Telemetry reflected power.
+		float tx_fwrdpwr	= 2; 	///< Tx Telemetry forward power.
+		float bus_volt		= 3; 	///< Tx Telemetry bus voltage.
+		float vutotal_curr	= 4; 	///< Tx Telemetry total current.
+		float vutx_curr		= 5; 	///< Tx Telemetry transmitter current.
+		float vurx_curr		= 6; 	///< Tx Telemetry receiver current.
+		float vupa_curr		= 7; 	///< Tx Telemetry power amplifier current.
+		float pa_temp		= 8; 	///< Tx Telemetry power amplifier temperature.
+		float board_temp	= 9; 	///< Tx Telemetry board temperature.
+		uint32 uptime		= 10;	///< Tx Uptime in Seconds.
+	}
+
+	receiverTelemetry receiver 			= 1;	///< Receiver Telemetry
+	transmitterTelemetry transmitter	= 2;	///< Transmitter Telemetry
+}
+
+// Information about the state of the Camera
+message CameraTelemetry {
+	uint32 upTime	= 1;	///< Local uptime (since last startup) in seconds
 }
 
+// Information about the state of the Electrical Power Supply
+message EpsTelemetry {
 
-// basic file transfer packet
-message FileTransferPacket {
-	int32 packetType = 1;	// the ID of the type of packet that is being sent (image, log, etc.)
-	int32 packetNumber = 2;	// the specific packet number that is being sent
-	bytes data = 3;			// the raw date bytes being transmitted
 }
 
+// Information about the state of the Battery
+message BatteryTelemetry {
+
+}
+
+// Information about the state of the Antenna
+message AntennaTelemetry {
+
+}
+
+
+// Dosimeter Payload Data (both boards)
+message DosimeterData {
+	// Payload Data for a single Dosimeter board
+	message DosimeterBoardData {
+		float voltageChannelZero	= 1;	///< COTS 2048mV REF IC, Shielding: none
+		float voltageChannelOne		= 2;	///< COTS 2048mV REF IC, Shielding: 50 mil
+		float voltageChannelTwo		= 3;	///< COTS 2048mV REF IC, Shielding: 100 mil
+		float voltageChannelThree	= 4;	///< COTS 2048mV REF IC, Shielding: 200 mil
+		float voltageChannelFour	= 5;	///< COTS 2048mV REF IC, Shielding: 20 mil
+		float voltageChannelFive	= 6;	///< RADFET Experimental Dosimeter, Shielding: none
+		float voltageChannelSix		= 7;	///< COTS 2048mV REF IC, Shielding: 300 mil
+		float voltageChannelSeven	= 8;	///< Temperature Sensor
+	}
+
+	DosimeterBoardData boardOne	= 1;	///< Payload Data from the first Dosimeter Board ("bottom" of the Satellite, beneath OBC)
+	DosimeterBoardData boardTwo	= 2;	///< Payload Data from the second Dosimeter Board ("top" of the Satellite, beneath Antenna)
+}
+
+// Image Packet
+message ImagePacket {

```

Comment:
I think we should include a frame index as part of the message as a backup in case things get jumbled.

Reply:
There's a frame ID within the image packet, is that what you mean? I was also planning on adding a packet sequence number to the header since we're considering allowing for windowed ACK-NAK

Reply:
That's exactly what I was thinking!